src/main/java/com/ldtteam/structurize/util/IOPool.java

```diff
@@ -0,0 +1,92 @@
+package com.ldtteam.structurize.util;
+
+import com.ldtteam.structurize.api.util.Log;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.concurrent.*;
+
+/**
+ * Class handling our IO pool.
+ */
+public final class IOPool
+{
+    /**
+     * Hold our IO queue.
+     */
+    private static final BlockingQueue<Runnable> ioQueue = new LinkedBlockingDeque<>();
+
+    /**
+     * Holds the specific executor for the queue.
+     */
+    private static ThreadPoolExecutor executor;
+
+    /**
+     * Structurize specific thread factory.
+     */
+    public static class StructurizeThreadFactory implements ThreadFactory
+    {
+        /**
+         * Ongoing thread IDs.
+         */
+        public static int id;
+
+        @Override
+        public Thread newThread(@NotNull final Runnable runnable)
+        {
+            final Thread thread = new Thread(runnable, "Structurize IO Worker #" + (id++));
+            thread.setDaemon(true);
+
+            thread.setUncaughtExceptionHandler((thread1, throwable) -> Log.getLogger().error("Structurize IO Thread errored! ", throwable));
+            return thread;
+        }
+    }
+
+    /**
+     * Creates a new thread pool for pathfinding jobs
+     *
+     * @return the threadpool executor.
+     */
+    public static ThreadPoolExecutor getExecutor()
+    {
+        if (executor == null)
+        {
+            executor = new ThreadPoolExecutor(1, 2, 10, TimeUnit.SECONDS, ioQueue, new StructurizeThreadFactory());
```

Comment:
the amount of threads should definitely be configurable, with zero meaning immediate on-same-thread execution